mitosheet/mitosheet/parser.py

```diff
@@ -412,50 +413,73 @@ def get_raw_parser_matches(
         formula: str,
         formula_label: Union[str, bool, int, float], # Where the formula is written,
         string_matches: List[Any],
-        df: pd.DataFrame
+        dfs: List[pd.DataFrame],
+        df_names: List[str],
+        sheet_index: int
     ) -> List[RawParserMatch]:
     """
     Returns a list of RawParserMatch, which are either column header matches
-    or matches on the index. 
+    or matches on the index or a sheet. 
 
     The parsing strategy generally can be described as follows:
     1. Match columns headers that are not followed by indexes (for backwards compatibility)
     2. Match column headers that are followed by indexes
+    3. Match sheet names that are followed by !
 
     (2) requires doing some reasoning about _what_ sort of index it is, 
     and therefore what sort of match is possible. Casting from the string
     to a parsed value is then done, and we can determine the row offset 
     of the match.
 
-    formula = "=A0"
+    formula = "=df 1!A0"
     formula_label = 1 (aka formula was written in B1)
     string_matches = []
     df = pd.DataFrame({'A': [1], 'B': [1]})
 
     We would get the matches: 
     {
-                        'type': '{HEADER}',
-                        'substring_range': (1, 1),
-                        'unparsed':'A',
-                        'parsed': 'A',
-                        'row_offset': 0
+        'type': '{SHEET}',
+        'substring_range': (1, 4),
+        'unparsed':'df 1',
+        'parsed': 'df 1'
+    }
+    {
+        'type': '{HEADER}',
+        'substring_range': (1, 1),
+        'unparsed':'A',
+        'parsed': 'A',
+        'row_offset': 0
     }
     {
-                        'type': '{INDEX}',
-                        'substring_range': (2, 2),
-                        'unparsed':'0',
-                        'parsed': 0,
-                        'row_offset': -1
+        'type': '{INDEX}',
+        'substring_range': (2, 2),
+        'unparsed':'0',
+        'parsed': 0,
+        'row_offset': -1
     }
 
     These would be returned in reverse, so that they were easy to work with.
     """
 
+    df = dfs[sheet_index]
     index = df.index
     column_headers: List[ColumnHeader] = df.columns.to_list()
 
     raw_parser_matches: List[RawParserMatch] = []
 
+    for sheet_index, sheet_name in enumerate(df_names):
+        if safe_contains(formula, f'{sheet_name}!', column_headers):
+            raw_parser_matches.append({
+                'type': '{SHEET}',
+                'substring_range': (formula.index(f'{sheet_name}!'), formula.index(f'{sheet_name}!') + len(sheet_name)),
+                'unparsed': f'{sheet_name}!',
+                'parsed': sheet_name,
+                'row_offset': 0
+            })
+
+            # Update to look at the column headers in the other sheet
+            column_headers = dfs[sheet_index].columns.to_list()
```

Comment:
I think this won't work if you have a formula with column headers from the original dataframe, and a cross sheet reference. It'll miss the column header references to the original dataframe (as it's no longer searching for those column headers). 

Maybe we can just add the `column_headers` to the list of column headers? And make sure to deduplicate? A bit of a hack, but seems like it would work.